onboarding-app/SecondViewController.swift

```diff
@@ -103,13 +103,16 @@ final class SecondViewController: UIViewController {
         firstLineText: String,
         secondLineText: String,
         imageSystemName: String,
-        onTap: Selector? = nil
+        showCounterOnTap: Bool = false
     ) {
+        let onTap = showCounterOnTap ? { [weak self] in
+            self?.present(CounterViewController(), animated: true)
+        } : nil
+        
```

Comment:
At first, I tried this instead:
```
let onTap = showCounterOnTap ? { [weak self] in self?.showCounterView() } : nil
```
And I kept the function on lines 125-128 (I did, however, remove `@objc` since it's no longer needed).

However, this still led to a memory leak. Is this because there is still a strong reference to `self.present()` within the body of `showCounterView()`?